### PR TITLE
KAFKA-15270 - Simple Consume Integration tests

### PR DESCRIPTION
**Pull request by lianetm**
Initial integration tests for simple consume, including reset/validate positions functionality.

Includes also minor fixes & unit test

### CODE REVIEW

File: core/src/test/scala/integration/kafka/api/BaseAsyncConsumerTest.scala [hunk 1]
```diff
@@ -62,4 +67,61 @@ class BaseAsyncConsumerTest extends AbstractConsumerTest {
...
+
+  @Test
+  def testSimpleConsume(): Unit = {
+    val numRecords = 10
+
+    val producer = createProducer()
+    val startingTimestamp = System.currentTimeMillis()
+    sendRecords(producer, numRecords, tp, startingTimestamp = startingTimestamp)
+
+    val consumer = createConsumer(configsToRemove = List(ConsumerConfig.GROUP_ID_CONFIG))
```
Comment (philipnee) [inline core/src/test/scala/integration/kafka/api/BaseAsyncConsumerTest.scala:79]: I think we actually want to use `createAsyncConsumer` no?

Comment (lianetm) [reply core/src/test/scala/integration/kafka/api/BaseAsyncConsumerTest.scala:79]: oh I see you introduced the `createAsyncConsumer`, but this `createConsumer` is modified to create a PrototypeAsyncConsumer too (so we have 2 ways of doing this). Let's align with @kirktrue and choose the preferred one, and I'll remove the other one.

Comment (philipnee) [reply core/src/test/scala/integration/kafka/api/BaseAsyncConsumerTest.scala:79]: ah great. I think this should work too.

**Review by philipnee — APPROVED** [review]
look good thanks!

**Pull request merged**